flag/guest/cluster/kubernetes/ingresscontroller/docker/docker.go

```diff
@@ -0,0 +1,6 @@
+package docker
+
+// Docker is a data structure to hold Docker image configuration flag.
+type Docker struct {
+	Image string
+}
```

Comment:
I think all `Docker.Image` are candidates to be thrown away. We aren't going to configure docker images per instalation basis, and for sure not per cluster. We learned this is simply unmaintainable. For the transition period we can use constants, but later those should be removed all together. See https://github.com/giantswarm/giantswarm/issues/2383

Reply:
Good that you brought this up. I hadn't noticed that ticket. This changes my direction again because there are removed fields (e.g. API AltNames) that `cluster-operator` requires for certification creation.

Reply:
I'd be fine with having things like altNames in const/var. I think we should just have `kubernetes,kubernetes.default,kubernetes.default.svc,kubernetes.default.svc.cluster.local` everywhere.

In general I would be careful with exposing something as a flag, and **extra** careful with exposing something in custom object. Having something as a const/var allows to have global bug fixes and changes w/o difficult migration. I believe settings should be exposed as they are needed.  Otherwise situation described in https://github.com/giantswarm/giantswarm/issues/2383 can become worse.

Reply:
Ok. Had a discussion with @xh3b4sd and end result is that I'll continue using `Cluster` object here, but commented to referenced ticket that at least `Kubernetes.API.AltNames` field should be preserved. I'm fine with changing `Docker` structure though. @kopiczko How would you suggest to deal with that docker image field right now? Keep it as it is for now and fix later or change it promptly to something else?

Reply:
Having altNames as const is ok for me. I'll try to pay attention to other fields as I progress with implementation. What about those docker types?

Reply:
I'm bit confused. 

> I'll continue using Cluster object here

What do you mean? Is this a part of this discussion?

I think both altNames and dockers images should be consts/vars. I think making them configurabe is unmaintainable. At some point we should remove them from cluster object (when we remove old resource versions from operators, and old k8scloudocnfig versions). They aren't used in newer resources and cloudconfigs anyway.

Reply:
Oh, sorry I didn't elaborate properly earlier and later kind of replied to my personal thinking. Apologies for that. When you referred to that issue where there are fields of `Cluster` to be removed, I noticed that e.g. `Kubernetes.API.AltNames` was part of it and thought that I cannot use it anymore. Then I had discussion with Tim which lead to decision to still use `Cluster` object because that's the only sane thing to do here.

If I understand the situation here correctly, I can already use const `altName` and `dockerImage` here? For other stuff, I'll use `Cluster` as it is now.

Reply:
Yes, correct. We can treat those fields as deprecated.